content/articles/how-to-perform-logging-for-golang-applications/index.md

```diff
@@ -0,0 +1,290 @@
+
+The concept of logging tries to solve this for you. It leaves a trail of breadcrumbs. So whenever something goes wrong, you can determine the cause at the right time and solve it before it messes up your application. Logging allows you to write your application status messages to files, databases, or other output streams. These messages contain information on which parts of your code have been executed and what problems may have arisen.
+
+### Why logging
+
+As a developer, logging plays a critical role in fixing an error in your application. It helps you figure out which section of your code is causing an issue. This essentially helps you debug your application by letting you track step by step execution of your application. There are several tools that we can use for debugging. However, logging messages is arguably one of the most efficient ways to debug your code.
+
+To help you prioritize which code issues need more attention, logging classified logging messages to different logging levels. A logging level acts as a message filter. These logging include Trace < Debug < Info < Warn < Error < Fatal, based on the priority order.
+
+Logging can be applied to almost any programming language. Let's dive in and see how we can set up some basic logging for the Go application.
+
+### Set up logging for Go apps
+
+First, you need to initialize a Go application on your computer. Thus you need to Go install. Check Golang installation in case you haven't installed it yet. Once installed, run `go version` to check if Go has been installed. Then create and navigate to a project folder and initialize Go using `go mod init go-logging-app`. This will set up ready to write some Go code.
+
+Go has several standard packages that allow you to set loggers in your application. However, Go has a logging package built for native Golang.
+
+To set up a basic log with Go, create a `main.go` at the root of your project folder. To get started doing simple logging, you can use a [log](https://pkg.go.dev/log) package to create a simple base logger. Go ahead and add the following to your `main.go` file.
+
+```go
+package main
+
+import (
+    "log"
+)
+
+func main() {
+    log.Printf("This is my first baci Golang")
+}
+```
+
+Then run your application using `go run main.go`. This will log a basic massage to your console, i.e., `2022/01/29 16:51:08 This is my first basic Golang`. By default, it shows the date and time this log was created and the message that this log generated. This is where logging comes in handy. It gives you a timestamp to reach the exact time this log was recorded.
```

Comment:
```suggestion
Then run your application using `go run main.go`. This will log a basic message to your console, i.e., `2022/01/29 16:51:08 This is my first basic Golang`. By default, it shows the date and time this log was created and the message that this log generated. This is where logging comes in handy. It gives you a timestamp to reach the exact time this log was recorded.
```